pkg/cloudprovider/launchtemplate.go

```diff
@@ -146,9 +150,21 @@ func (p *LaunchTemplateProvider) createAmiOptions(ctx context.Context, nodeTempl
 	if len(securityGroupsIDs) == 0 {
 		return nil, fmt.Errorf("no security groups exist given constraints")
 	}
+
+	clusterEndpoint := awssettings.FromContext(ctx).ClusterEndpoint
```

Comment:
I don't think we want to do the discovery like this. Ideally, we encapsulate the fact that we are discovering this behavior so that we resolve it once in `settings.Settings`. That way, any call to `settings.FromContext(ctx).ClusterEndpoint` just returns the cluster endpoint, regardless of whether it was discovered, or user-given

Reply:
If we move this to `settings` wouldn't it hide the fact that setting is something else (empty)?

Because we only require the `clusterEndpoint` to populate the launch template and for nothing else, wouldn't this be the best place to only fetch that here?

Reply:
No, the settings is whatever the resolved value is. We shouldn't try to keep what is actually passed through. The same thing happens with our defaults. If a user doesn't set something, then the resolved value ends up being put into the settings. 

I'm trying to protect against future uses of `settings.FromContext(ctx).ClusterEndpoint` which return empty, which is going to be surprising to devs if it's not the same everywhere.

Reply:
Then I need some guidance please from you, I'm not exactly sure, how I should pass the API endpoint seamlessly then, to fit with the existing structure.